src/Host/Monitor/Impl/BrokerManager.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.R.Host.Monitor {
+    public class BrokerManager {
+        private const string RHostBroker = "Microsoft.R.Host.Broker";
+        private static string RHostBrokerExe = $"{RHostBroker}.exe";
+        private static string RHostBrokerconfig = $"{RHostBroker}.Config.json";
+        private static Process _brokerProcess;
+       
+        public static bool AutoRestart { get; set; }
+        public static int AutoRestartCount {
+            get {
+                return Properties.Settings.Default.AutoRestartCount;
+            }
+            set {
+                Properties.Settings.Default.AutoRestartCount = value;
+            }
+        }
+
+        private static int _autoRestartCount = 0;
```

Comment:
It looks like this isn't the backing field for `AutoRestartCount`. If so, it shouldn't have the same name, to avoid confusion.